L1Trigger/TrackFindingTracklet/src/CandidateMatchMemory.cc

```diff
@@ -0,0 +1,67 @@
+#include "L1Trigger/TrackFindingTracklet/interface/CandidateMatchMemory.h"
+#include "L1Trigger/TrackFindingTracklet/interface/Settings.h"
+#include "L1Trigger/TrackFindingTracklet/interface/Tracklet.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+using namespace std;
+using namespace Trklet;
+
+CandidateMatchMemory::CandidateMatchMemory(string name, const Settings* const settings, unsigned int iSector)
+    : MemoryBase(name, settings, iSector) {}
+
+void CandidateMatchMemory::addMatch(std::pair<Tracklet*, int> tracklet, std::pair<Stub*, L1TStub*> stub) {
+  std::pair<std::pair<Tracklet*, int>, std::pair<Stub*, L1TStub*> > tmp(tracklet, stub);
+
+  //Check for consistency
+  for (unsigned int i = 0; i < matches_.size(); i++) {
+    if (tracklet.first->TCID() < matches_[i].first.first->TCID()) {
+      edm::LogPrint("Tracklet") << "In " << getName() << " adding tracklet " << tracklet.first
+                                << " with lower TCID : " << tracklet.first->TCID() << " than earlier TCID "
+                                << matches_[i].first.first->TCID();
+      assert(0);
+    }
+  }
+  matches_.push_back(tmp);
+}
+
+void CandidateMatchMemory::writeCM(bool first) {
+  std::string fname = "../data/MemPrints/Matches/CandidateMatches_";
```

Comment:
see above

Reply:
implemented as suggested (for all instances)